### PR TITLE
add petlisting pages (view, create, edit)

### PR DESCRIPTION
**Pull request by JuneOh710**
UI:
- with images added
image

- with no images
Screenshot 2023-12-03 at 5 17 32 PM

- the "edit post" btn only shows up for the pet listing owner(shelter1@example.com). 
- upon failed request to backend, render the error message for now.

### CODE REVIEW

**Review by lujorryn — APPROVED** [review]

File: frontend/src/pages/petlistings/PetDetail.jsx [hunk 1]
```diff
@@ -1,8 +1,244 @@
...
+        )
+      }
+      
+      return (
+        // returning:
+        // contact shelter btn
+        // application form btn
+        <>
+        <Button classes="square-btn" id="contact-shelter-btn" handleClick={() => {navigate(shelterContact)}}>contact shelter</Button>
+        <Button id="application-form-btn" handleClick={() => {navigate(`/applications`)}}>Application Form</Button>
```
Comment (lujorryn) [inline frontend/src/pages/petlistings/PetDetail.jsx]: Screen Shot 2023-12-04 at 6 51 28 PM

For shelter users these 2 buttons should not appear on other shelters' petlistings

**Review by lujorryn — CHANGES_REQUESTED** [review]

File: frontend/src/pages/petlistings/PetDetailCreate.jsx [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+                                        <label htmlFor="gender" className="key">Sex:</label>
+                                        <input name="gender" className="value" value={listing.gender} onChange={handlePetChange} />
+                                    </div>
+                                    <div className="row">
+                                        <label htmlFor="pet-status" className="key">Status:</label>
+                                        <input id="pet-status" className="value" name="status" value={listing.status} onChange={handlePetChange} />
+                                    </div>
+                                    <div className="row">
+                                        <label htmlFor="category" className="key">Category:</label>
+                                        <input id="category" className="value" name="category" value={listing.category} onChange={handlePetChange} />
```
Comment (lujorryn) [inline frontend/src/pages/petlistings/PetDetailCreate.jsx]: Can these fields (plus the size one above) be converted into drop downs instead or have the options given inside the placeholder ? It is slightly confusing to use if I don't know what's the fix set that I should be entering

**Review by lujorryn — APPROVED** [review]

File: frontend/src/pages/petlistings/PetDetailCreate.jsx [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+        if (Array.isArray(photos)) {
+            const photoUrls = photos.map((photo) => {
+                return `${process.env.REACT_APP_API_URL}/${photo.url}`;
+            });
+            return photoUrls;
+        }
+        return [defaultPhoto];
+    }
+
+    const stringToCode = (key) => {
```
Comment (JuneOh710) [inline frontend/src/pages/petlistings/PetDetailCreate.jsx:122]: parser thingy

Comment (lujorryn) [reply frontend/src/pages/petlistings/PetDetailCreate.jsx:122]: If the user enters something like 'puppy' into the category due to them not knowing that dog / cat are what you're supposed to enter, then puppy will become 'Other'. So I'd say even with the parser having a placeholder hint would be nice.

**Comment by lujorryn** [comment]
Regarding the Pet Status Field:

Create
I'm not sure if this is true but iirc a pet is 'available' on creation? If so then on the PetCreate page there shouldn't even be a status field.

Edit
I think this would work better as a radio if it's not too much trouble to change, because having unknown status default to pending sounds a bit weird to me.

File: frontend/src/pages/petlistings/PetDetailCreate.jsx (hunk 2 above)
Comment (JuneOh710) [reply frontend/src/pages/petlistings/PetDetailCreate.jsx]: implemented as radio

**Pull request merged**